lib/facility.rb

```diff
@@ -1,14 +1,76 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def has_service?(service)
+    @services.include?(service)
+   end
+
+  def allow_service(service)
+    if has_service?(service)
+      yield
+    else
+      false
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if 
+      @registered_vehicles << vehicle
+      vehicle.set_date
+      vehicle.give_plate
+      if vehicle.plate_type == :ev
+       @collected_fees += 200
+      elsif vehicle.plate_type == :antique
+        @collected_fees += 25
+      else
+        @collected_fees += 100
+      end
+      return [vehicle]
+    else 
+      return false
+    end
+  end
+
+  def administer_written_test(registrant)
+    return false unless registrant.age > 15 && registrant.permit?
+
+    allow_service('Written Test') do
```

Comment:
I'd rather just have you check this condition on line 53 with an if statement. So line 53 would say: `if registrant.age > 15 && registrant.permit? && has_service("Written Test")` and then put in that if block the logic of what to do if all those conditions are true. Then we only have to read one line to know what's going on, keeping all the conditions in one place.

Reply:
I thought about that but it felt lie a really long line

Reply:
That's true, but you can actually stack them vertically to make it not a long line ;) So after each logical operator, put the next condition on the next line